addons/payment_authorize/models/payment.py

```diff
@@ -276,12 +276,12 @@ def _authorize_s2s_validate(self, tree):
                     'date_validate': fields.Datetime.now(),
                 })
             if tree.get('x_type').lower() == 'auth_only':
-                if self.callback_eval:
-                    safe_eval(self.callback_eval, {'self': self})
                 self.write({
                     'state': 'authorized',
                     'acquirer_reference': tree.get('x_trans_id'),
                 })
+                if self.callback_eval:
+                    safe_eval(self.callback_eval, {'self': self})
```

Comment:
you were right: the code in the previous code block can be moved as well (move callback eval after writing on tx)